### PR TITLE
Finished Bootcamp

### PR DESCRIPTION
**Pull request by KarthiU**

### CODE REVIEW

File: modules/bootcamp/detect_landing_pad.py [hunk 1]
```diff
@@ -84,29 +84,38 @@ def run(self, image: np.ndarray) -> "tuple[list[bounding_box.BoundingBox], np.nd
...
+        predictions = self.__model.predict(source=image, conf=0.7, device=self.__DEVICE, verbose=False)
 
         # Get the Result object
-        prediction = ...
+        prediction = predictions[0]
 
         # Plot the annotated image from the Result object
         # Include the confidence value
-        image_annotated = ...
+        image_annotated = prediction.plot()
```
Comment (Xierumeng) [inline modules/bootcamp/detect_landing_pad.py:95]: Are the confidence labels included in the logged images?

File: modules/bootcamp/detect_landing_pad.py [hunk 2]
```diff
@@ -84,29 +84,38 @@ def run(self, image: np.ndarray) -> "tuple[list[bounding_box.BoundingBox], np.nd
...
         # and convert to a numpy array
-        boxes_cpu = ...
-
+        boxes_cpu = boxes_xyxy.detach().cpu().numpy()
+    
         # Loop over the boxes list and create a list of bounding boxes
         bounding_boxes = []
+
+        for i in range(boxes_cpu.shape[0]):
+            box = bounding_box.BoundingBox.create(boxes_cpu[i])[1]
```
Comment (Xierumeng) [inline modules/bootcamp/detect_landing_pad.py:109]: What if box is `None` ?

File: modules/bootcamp/decision_simple_waypoint.py [hunk 3]
```diff
@@ -37,7 +40,9 @@ def __init__(self, waypoint: location.Location, acceptance_radius: float):
         # ↓ BOOTCAMPERS MODIFY BELOW THIS COMMENT ↓
         # ============
 
-        # Add your own
+        
+        self.waypoint.location_x = waypoint.location_x
+        self.waypoint.location_y = waypoint.location_y
```
Comment (Xierumeng) [inline modules/bootcamp/decision_simple_waypoint.py:45]: This doesn't actually do anything, since the waypoint was already set above.

File: modules/bootcamp/decision_simple_waypoint.py [hunk 4]
```diff
@@ -68,10 +73,24 @@ def run(self,
         # ↓ BOOTCAMPERS MODIFY BELOW THIS COMMENT ↓
         # ============
 
+        if report.status == drone_status.DroneStatus.HALTED: 
+            current_x = report.position.location_x
+            current_y = report.position.location_y
+            relative_y = self.waypoint.location_y - current_y
+            relative_x = self.waypoint.location_x - current_x
+
+            radius_sq = pow(relative_y, 2) + pow(relative_x, 2)
```
Comment (Xierumeng) [inline modules/bootcamp/decision_simple_waypoint.py:82]: Nice.

Comment (Xierumeng) [inline modules/bootcamp/decision_simple_waypoint.py:82]: `pow()` is less readable than the `**` operator. (You do not need to change anything here, just something to think about).

File: modules/bootcamp/detect_landing_pad.py [hunk 5]
```diff
@@ -84,32 +84,43 @@ def run(self, image: np.ndarray) -> "tuple[list[bounding_box.BoundingBox], np.nd
...
-        boxes_xyxy = ...
+        boxes_xyxy = prediction.boxes.xyxy
 
         # Detach the xyxy boxes to make a copy,
         # move the copy into CPU space,
         # and convert to a numpy array
-        boxes_cpu = ...
-
+        boxes_cpu = boxes_xyxy.detach().cpu().numpy()
+
```
Comment (Xierumeng) [inline modules/bootcamp/detect_landing_pad.py:104]: Nit: Avoid leaving whitespace when it's not necessary. (You do not need to change anything here, just something to think about).

File: modules/bootcamp/decision_waypoint_landing_pads.py [hunk 6]
```diff
@@ -70,8 +79,42 @@ def run(self,
...
+            relative_y = self.waypoint.location_y - current_y
+            relative_x = self.waypoint.location_x - current_x
+          
+            radius_sq = pow(relative_y, 2) + pow(relative_x, 2) 
+
+            if radius_sq <= pow(self.acceptance_radius, 2): 
+                if self.waypoint_flag == True: 
+                    command = commands.Command.create_land_command()
+                else: 
+                    closest_pad = 100000000
```
Comment (Xierumeng) [inline modules/bootcamp/decision_waypoint_landing_pads.py:98]: What if the landing pad's distance could be arbitrarily large (e.g. all measurements are changed to be in nanometres)? There exists a value that is guaranteed to be larger that all other values.

File: modules/bootcamp/decision_waypoint_landing_pads.py [hunk 7]
```diff
@@ -70,8 +79,42 @@ def run(self,
...
+            relative_y = self.waypoint.location_y - current_y
+            relative_x = self.waypoint.location_x - current_x
+          
+            radius_sq = pow(relative_y, 2) + pow(relative_x, 2) 
+
+            if radius_sq <= pow(self.acceptance_radius, 2): 
+                if self.waypoint_flag == True: 
+                    command = commands.Command.create_land_command()
+                else: 
+                    closest_pad = float('inf')
```
Comment (Xierumeng) [inline modules/bootcamp/decision_waypoint_landing_pads.py:98]: Prefer double quotation marks for string literals. Single quotation marks are for character literals.

**Review by Xierumeng — COMMENTED** [review]
LGTM.
